api/src/test/java/com.codeforcommunity/rest/PostsRouterTest.java

```diff
@@ -0,0 +1,173 @@
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.codeforcommunity.api.IPostsProcessor;
+import com.codeforcommunity.dto.response.PostSummary;
+import com.codeforcommunity.dto.response.PostsResponse;
+import com.codeforcommunity.dto.response.SinglePostResponse;
+import com.codeforcommunity.rest.PostsRouter.Externals;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.Route;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+/**
+ * This class is to test the router. Usually, we don't end up testing the router, but we think it
+ * would be pretty useful to show you the Mockito dependency. Because this is purely for that, we'll
+ * only add a couple examples.
+ *
+ * <p>Mockito is pretty cool, since it allows you to define what happens and what's returned/thrown
+ * when a method is called. It also lets you verify methods are called (with specific params if you
+ * want!), capture those params for testing, spy on existing objects so you can define the bare
+ * minimum if you want, and some other interesting functionality. Go check it out!
+ *
+ * <p>Check out the Mockito documentation for more info
+ * https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html.
+ */
+public class PostsRouterTest {
+  private PostsRouter router;
+  private IPostsProcessor processor;
+  private Vertx vertx;
+  private Router vertxRouter;
+  private Route route;
+  private RoutingContext ctx;
+  private HttpServerResponse res;
+
+  @BeforeEach
+  public void setup() {
+    // The mock method allows us to create a fake version of the Vertx class (which we usually don't
+    // have a lot
+    // of control over) where we can simply write the interactions we want it to have.
+    this.vertx = mock(Vertx.class);
+    // Remember, we don't have access to this since the api module doesn't have a dependency on the
+    // service module (that would create a dependency cycle, which is really bad).
+    this.processor = mock(IPostsProcessor.class);
+    this.router = new PostsRouter(this.processor, new TestExternals());
+    this.vertxRouter = mock(Router.class);
+    this.route = mock(Route.class);
+    this.ctx = mock(RoutingContext.class);
+
+    // Mock for the IRouter#end method. We don't have to handle HttpServerResponse#end method, since
+    // unmocked methods just return null.
+    res = mock(HttpServerResponse.class);
+    when(res.setStatusCode(anyInt())).thenReturn(res);
+    when(res.putHeader(anyString(), anyString())).thenReturn(res);
+    when(ctx.response()).thenReturn(res);
```

Comment:
maybe a little annotation if you think it is necessary rather then in a big paragraph